example-client/README.md

```diff
@@ -3,11 +3,219 @@ Example CoAP client
 
 This is an example coap client application, that can be use with any LwM2M Server.
 
+# Running
+
 Usage
 -----
 
     ./run.sh <registration-url> [<keystore-file>]
 
+note: parameters are parsed like a [www URL query string](https://en.wikipedia.org/wiki/Query_string) where ? and & are delimeters for the endpoint name and account ID
+
 *For example:*
-        
-    ./run.sh 'coap://localhost:5683/rd?ep=device007&lt=3600&b=U'    
+    
+    cd java-coap/example-client
+
+    ./run.sh "coaps://lwm2m.us-east-1.mbedcloud.com:5684/rd?ep=${ENDPOINT_NAME}&aid=${ACCOUNT_ID}" ../../keystore/example-client-device-keystore.jks
+
+<img src="images/example-running.png" width="600">
+
+    *note: replace ${ENDPOINT_NAME} and ${ACCOUNT_ID} with the value defined in the Configuring section below*
+
+*${ENDPOINT_NAME} can be obtained by running the following command*
+```
+keytool -list -v -storepass secret -keystore example-client-device-keystore.jks 2> /dev/null | grep 'Alias name: mbed-cloud'
+```
+
+<img src="images/example-connected.png" width="600">
+
+# Building with Maven
+
+#### Clone java-coap
+
+```
+git clone https://github.com/ARMmbed/java-coap
+```
+
+### Change to the project
+
+```
+cd java-coap
+```
+
+### Maven install the project
+
+```
+mvn package install -DskipTests
+```
+
+### Change to the example client
+
+```
+cd example-client
+```
+
+# Configuring
+
+## Generating Keystore, Certificates, Security Keys
+
+### Set these variables which can be retrieved from https://portal.us-east-1.mbedcloud.com/account
```

Comment:
- Too long headings. 
- As above, recommending to use numbered steps
- Links to portal without showing the URL:  ... from the [Mbed Cloud portal](write the URL here)-